content/guides/concepts/dht.md

```diff
@@ -0,0 +1,85 @@
+- answered with contacting information of a node having better chances to be able to answer (IPFS uses this strategy)
+
+DHT's decentralization provides advantages compared to a classic Key-Value store:
+- *scalability* as a request for a hash of length *n* takes at most *log2(n)* steps to resolve.
+- *fault tolerance* via redundancy, so that lookups are possible even if peers unexpectedly leave or join the DHT. Additionally, requests can be addressed to any peer if one is slow or unavailable.
+- *load balancing* as requests are made to different nodes and no unique peers process all the requests. 
+
+## How do DHTs work?
+
+### Peer IDs
+Each peer have a peerID which is a hash with the same length *n* as the DHT keys.
+
+### Buckets
+A subset of the DHT maintained by a peer is called a 'bucket'. 
+A bucket maps to hashes with the same prefix as the PeerID up to *m* bits. There are 2^m buckets. Each bucket maps for 2^(n-m) hashes.
+
+- For example, if m = 2^16 and with an hexadecimal encoding (4 bits per displayed character), the peer with PeerID "ABCDEF12345" maintains mapping for hashes starting with "ABCD". 
+Some hashes falling into this bucket would be *ABCD*38E56, *ABCD*09CBA or *ABCD*17ABB ... 
+
+The size of the buckets are related to the size of the prefix. The longer the prefix, the less hashes each peer has to manage, the more peers are needed.
+Several peers can be in charge of the same bucket if they have the same prefix.
+
+In most DHTs, including IPFS's Kademlia implementation, the size of the buckets (and the size of the prefix), are dynamic. 
+Buckets size growths and prefix size shortens when many nodes leaves the DHT, and vice versa. **(/!\ or does it depends on the number of records and not the number of nodes? Is it at bucket level or DHT level?)**
+
+### Peer lists
+
+Peers also keep connection to other peers to forward requests if the requested hash is not in their own bucket.
+
+If hashes are of length n, a peer will keep n-1 lists of peers: 
+- the first list contains peers which ID have a different 1st bit.
+- the second list contains peer which have their first bits identical to its own, but a different second bit
+- ...
+- the m-th list contains peer which have their first m-1 bits identical, but a different m-th bit
+- ...
+
+The higher m, the harder it is to find peers which have the same ID up to m bits. The lists of "closest" peers typically remains empty.
+"Close" here is defined as the XOR distance, so the longer the prefix they share, the closer they are.
+Lists also have a maximum of entries k, otherwise the first lists would contain half the network, then a fourth, etc.
+
+### Using the DHT
+
+When a peer receives a lookup request, it will either answer with a value if it falls into its own bucket, or answer with the contacting information (IP + port, peerID, etc) of a closer peer. The requesting peer can then send its request to this closer peer. The process goes on until a peer is able to answer it. 
```

Comment:
```suggestion
When a peer receives a lookup request, it will either answer with a value if it falls into its own bucket, or answer with the contacting information (IP+port, `peerID`, etc.) of a closer peer. The requesting peer can then send its request to this closer peer. The process goes on until a peer is able to answer it. 
```

Reply:
Moved this change to https://github.com/ipfs/docs/pull/364